### ISSUE TITLE
retries broken on dotnet-install.sh 

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/arcade#6761]
```
2021-01-07T23:30:53.9907369Z ##[section]Starting: Restore and Build Product
2021-01-07T23:30:53.9917230Z ==============================================================================
2021-01-07T23:30:53.9917870Z Task         : Command line
2021-01-07T23:30:53.9918442Z Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
2021-01-07T23:30:53.9918959Z Version      : 2.178.0
2021-01-07T23:30:53.9919370Z Author       : Microsoft Corporation
2021-01-07T23:30:53.9919930Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
2021-01-07T23:30:53.9920574Z ==============================================================================
2021-01-07T23:30:54.3431532Z Generating script.
2021-01-07T23:30:54.3439495Z Script contents:
2021-01-07T23:30:54.3440578Z ./build.sh -subset libs+libs.tests -rc release -configuration Debug -ci -arch x64 -framework net6.0  -testscope innerloop      /p:ArchiveTests=true
2021-01-07T23:30:54.3441325Z ========================== Starting Command Output ===========================
2021-01-07T23:30:54.3460121Z [command]/usr/bin/bash --noprofile --norc /__w/_temp/cf3903db-f5db-47ff-a732-8d1d2b5c7a2e.sh
2021-01-07T23:30:54.3789136Z __DistroRid: linux-x64
2021-01-07T23:30:54.4143073Z Downloading 'https://dot.net/v1/dotnet-install.sh'
2021-01-07T23:30:54.6257921Z curl: (35) Network file descriptor is not connected
2021-01-07T23:30:54.6265950Z Curl failed; dumping some information about dotnet.microsoft.com for later investigation
2021-01-07T23:30:54.6270188Z /__w/1/s/eng/common/tools.sh: line 277: openssl: command not found
2021-01-07T23:30:54.6327541Z ##[error]Bash exited with code '127'.
2021-01-07T23:30:54.6630932Z ##[section]Finishing: Restore and Build Product
```

this line needs to still go ahead and retry even if openssl isn't installed

https://github.com/dotnet/arcade/blob/master/eng/common/tools.sh#L277

**Comment by danmosemsft** [comment]
@MattGal looks like your change https://github.com/dotnet/arcade/commit/6425f16995bad0d288da00d865b75a86512d22a0 unfortunately broke retries, at least on whatever we're building our product on.

Not sure whether we want to check for openssl existing, or just install it.

**Comment by safern** [comment]
> Not sure whether we want to check for openssl existing, or just install it.

I would say let's check if openssl exists and if it doesn't just skip the step that dumps extra info and go right into retries.

**Comment by MattGal** [comment]
@danmosemsft sorry about that I'll get that going; shouldn't be terribly critical as it's only in the failure path.  Amazing that this log shows _another_ way for curling that script to fail.

**Comment by wfurt** [comment]
we can argument that to simply ignore failures.  e.g. `|| true`
May be easier than the detection - and this should be temporarily anyway until we get root cause

**Comment by danmosemsft** [comment]
No problem thanks @MattGal !

**Comment by MattGal** [comment]
https://github.com/dotnet/arcade/pull/6762

**Comment by ViktorHofer** [comment]
Fixed by #6762
